InteropIO/finsemble-seed
# [17188] [FOR REVIEW ONLY] : Data Migration Recipe

[Kanbanize](https://chartiq.kanbanize.com/ctrl_board/21/cards/17188/details/)

Recipe for migration of user data.  This example migrates from IndexedDB to LocalStorage for the `finsemble` and `finsemble.workspace` topics. The developer may add or subtract topics and change adapters as per their needs. 

**Setup**
1. [Enable auth](https://documentation.chartiq.com/finsemble/tutorial-Authentication.html) in a local instance from a `release` or `master` branch (_i.e._, not this branch).
1. Start Finsemble.
1. Log in and create workspace(s), launch components, etc.
1. Save workspace(s).
1. Quit Finsemble.

**Testing**
1. Check out branch.
1. [Enable auth](https://documentation.chartiq.com/finsemble/tutorial-Authentication.html).
1. Start Finsemble.
1. Log in with previous user.
1. Migration Assistant component should spawn.
1. Click `Begin migration now.`
1. Migration should occur and present with a `Restart to continue.` button.
1. Click `Restart to continue.`.
1. Log in with previous user.

**Expected Result**: workspace is intact, no migration component spawns.
